Adds Metamask in the Ethereum API Section (#31)
moondance-labs/tanssi-docs: builders/interact/ethereum-api/wallets/metamask.md

---
title: How to Connect MetaMask
description: This guide walks you through how to connect MetaMask, a browser-based Ethereum wallet, to your Tanssi EVM ContainerChain and how to transfer funds.
---

# Interacting with your Tanssi EVM ContainerChain Using MetaMask

## Introduction {: #introduction } 

> Developers building dApps on top of Tanssi EVM ContainerChains can leverage their Ethereum compatibility features integrate known Ethereum wallets, such as [MetaMask](https://metamask.io/){target=_blank}. By doing so, they can use the injected library MetaMask provides to interact with the Tanssi EVM ContainerChain.

Author: I think we normally say dApp instead of Dapps, will fix the other one

This guide takes you through all the necessary steps: from installing Metamask, to setting up a wallet, and finally connecting it to your Tanssi EVM ContainerChain.

## Install the MetaMask Extension {: #install-the-metamask-extension } 

> First, you'll start with a fresh and default [MetaMask](https://metamask.io/){target=_blank} installation from the Chrome store. After downloading, installing, and initializing the extension, follow the **Get Started** guide. In there, you need to create a wallet, set a password, and store your secret backup phrase (this gives direct access to your funds, so make sure to store these in a secure place). 

Reviewer: ```suggestion
First, you'll start with a fresh and default [MetaMask](https://metamask.io/){target=_blank} installation from the Chrome store. After downloading, installing, and initializing the extension, follow the **Get Started** guide it will present. In there, you need to create a wallet, set a password, and store your secret backup phrase (this gives direct access to your funds, so make sure to store these in a secure place). 
```

> 

Reviewer: I'd add a note like this in the install section

```suggestion
!!! note
    The Metamask browser extension is compatible with Chrome, Chromium based browsers (such as Microsoft Edge and Opera), and Firefox. Metamask is also available as a mobile app for iOS and Android devices.
```

## Setup a Wallet {: #setup-a-wallet } 

> After installing [MetaMask](https://metamask.io){target=_blank}, the setup will automatically open a new task with a welcome screen. Here, you are oferred two options:

Reviewer: ```suggestion
After installing [MetaMask](https://metamask.io){target=_blank}, the setup will automatically open a new task with a welcome screen. Here, you are offered two options:
```

Here, you can configure MetaMask for the following networks:

|         Variable          |                           Value                            |
|:-------------------------:|:----------------------------------------------------------:|
|       Network Name        |               `EVM ContainerChain Dancebox`                |
|          RPC URL          | `https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network` |
|         Chain ID          |                           `5678`                           |
|     Symbol (Optional)     |                           `UNIT`                           |
| Block Explorer (Optional) |                           `N/A`                            |

To do so, fill in the following information:

1. **Network name** - name that represents the network you are connecting to
2. **RPC URL** - RPC endpoint of the network 
3. **Chain ID** - Chain ID
4. **Symbol** - (optional) symbol of the native token of the network
5. **Block Explorer** - (optional) URL of the block explorer
6. Once you've verified all the information, click on **Save**

![Add network in Metamask](/images/builders/interact/ethereum-api/wallets/metamask/metamask-8.png)

> Once you've added the network, you'll be redirected to a screen stating that you've successfully added a network. Furthermore, you'll be prompted a button to **Switch to Moonbase Alpha**, the network added in this example.

Reviewer: ```suggestion
Once you've added the network, you'll be redirected to a screen stating that you've successfully added a network. Furthermore, you'll be prompted a button to **Switch to EVM ContainerChain Dancebox**, the network added in this example.
```
